nexus/types/src/external_api/params.rs

```diff
@@ -150,6 +176,140 @@ impl InstanceSelector {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, PartialEq)]
+pub struct OptionalInstanceSelector {
+    #[serde(flatten)]
+    pub instance_selector: Option<InstanceSelector>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, PartialEq)]
+pub struct NetworkInterfaceSelector {
```

Comment:
I'm sure it's annoying and duplicative, but it would personally really help to have docstrings on these. Since those are likely to be nearly the same for all these `*Selector` types, it'd be enough to have a good docstring on one and refer to it in the others. A block comment in the code describing what these mean would be fine too.

Reply:
Ah, yeah, I didn't mean for OpenAPI consumers, but for developers working on Omicron itself. I realize that's pretty annoying since you've written it all and know exactly what everything means, so I definitely think the block-comment-and-references approach is sufficient!

Reply:
Ah, no, that makes sense. I didn't understand what you were saying in the beginning. I'm going to take another pass in this file post the V1 migration and I'll make sure there are good docs for everything here.